api/filters/replacement/replacement.go

```diff
@@ -142,9 +142,6 @@ func applyToOneNode(options *types.FieldOptions, t *yaml.RNode, value *yaml.RNod
 	}
 
 	for _, scalarNode := range t.YNode().Content {
-		if options != nil && options.Create {
-			return fmt.Errorf("cannot use create option in a multi-value target")
-		}
```

Comment:
It's also true that the current implementation is incompatible with Create mode though, and the error thrown when you try to enable it is nasty. For example, try updating `list index contains '*' character` test by adding the create option (no other changes). The error is ` wrong Node Kind for spec.template.spec.containers expected: MappingNode was SequenceNode: value: {- image: nginx`. We should fix the feature or the error, but I don't think either is easy to do.

The culprit is continued PathGetter use at L120, since PathGetter does not handle wildcards--that wouldn't especially make sense, since it can only return one result. It's not a simple fix though, since PathMatcher conversely does not handle creation--that wouldn't especially make sense, since its purpose it to retrieve existing paths.

Both the escape clause above and the nil check at L127 are somewhat adding to the confusion, because those conditions can only happen in one path other the other. I'd suggest calling `setTargetValue` straight from the Create section, and collapsing `applyToOneNode` into the non-create section, where we know we don't need to handle the nil or empty content cases.

Reply:
> I tried running your regression test, and it does not pass in the commit that introduces it--it generates the desired yaml just fine. Maybe you meant to use a job that already has the targeted containers path?

Ah you're right, I updated the test with the actual contents of the bug report. It throws an error if it already contains the targeted path and `create: true` is set. 

> what was the original reason for the constraint?

The constraint was intended to be that if there are multiple values targetted by a fieldpath, e.g. by using the wildcard match, we should not support `create: true`. The author noted that there were no use cases to support creating nodes for wildcard matches and that `PathMatcher` didn't support handling creation. I agree with the latter point, but the constraint where it is implemented is now throwing errors where it shouldn't, and not actually being hit where it should (you're getting `wrong Node Kind for spec.template.spec.containers expected: MappingNode was SequenceNode: value: {- image: nginx` when you should be getting `cannot use create option in a multi-value target`).

The error message you've ran into is also a bug report here: https://github.com/kubernetes-sigs/kustomize/issues/4561. This PR actually doesn't change that behavior, but I think I can write a check to ensure we throw the right error message for that.  

> As a sidenote, I found this code path hard to follow, in part because of the naming. applyToNode calls applyToOneNode, and neither describes what the function actually does accurate. The former is given an entire target rather than a terminal node (to be fair, everything is a node), and the latter specifically mutates a list of terminal nodes, not one. The previous PR switched us from digging through the target with a PathGetter, which returns one node, to a PathMatcher, which ALWAYS returns a sequence node. applyToOneNode knows that, since it always iterates if there was any content at all. But its name is very misleading.

I'll take some time to refactor the code to make it easier to follow while providing a fix for the bug.